### PR TITLE
decision and subflow chages

### PR DESCRIPTION
**Pull request by abilpraju-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-3760
https://aottech.atlassian.net/browse/FWF-3836
https://aottech.atlassian.net/browse/FWF-3810
https://aottech.atlassian.net/browse/FWF-3837

### CODE REVIEW

File: forms-flow-web/src/apiManager/endpoints/index.js [hunk 1]
```diff
@@ -85,7 +85,9 @@ const API = {
   VALIDATE_TENANT: `${MT_ADMIN_BASE_URL}/${MT_ADMIN_BASE_URL_VERSION}/tenants/<tenant_id>/validate`,
   VALIDATE_FORM_NAME: `${WEB_BASE_URL}/form/validate`,
   EXPORT_FORM: `${WEB_BASE_URL}/form/<form_id>/export`,
-  GET_PROCESS_XML: `${WEB_BASE_URL}/process/key/<process_key>`
+  GET_PROCESS_XML: `${WEB_BASE_URL}/process/key/<process_key>`,
+  PUBLISH_BPMN: `${WEB_BASE_URL}/process/<process_id>/publish`,
+  UN_PUBLISH_BPMN: `${WEB_BASE_URL}/process/<process_id>/unpublish`
```
Comment (arun-s-aot) [inline forms-flow-web/src/apiManager/endpoints/index.js:90]: is this endpoint common for BPMN & DPMN ? @abilpraju-aot

Comment (auslin-aot) [reply forms-flow-web/src/apiManager/endpoints/index.js:90]: For both dmn & bpmn same endpoint

Comment (arun-s-aot) [reply forms-flow-web/src/apiManager/endpoints/index.js:90]: Please give a general name ( like PUBLISH_PROCESS , UNPUBLISH_PROCESS)

File: forms-flow-web/src/components/Modeler/SubflowCreateEdit.js [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+          </div>
+        </Card.Body>
+      </Card>
+
+      <Card>
+        <div className="wraper">
+          <Card.Header>
+            <div
+              className="d-flex justify-content-between align-items-center"
+              style={{ width: "100%" }}
```
Comment (fahad-aot) [inline forms-flow-web/src/components/Modeler/SubflowCreateEdit.js:375]: Here also

File: forms-flow-web/src/components/Modeler/SubflowCreateEdit.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+      if (!isPublished && !validateProcess(xml, lintErrors)) {
+        return;
+      }
+
+      const actionFunction = isPublished ? unPublish : publish;
+
+      const response = !isPublished ? await saveFlow(!isPublished) : null;
+
+      closeModal();
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modeler/SubflowCreateEdit.js:171]: will it be good to call closeModal after API call response is received? @abilpraju-aot

**Comment by arun-s-aot** [comment]
SonarCloud warning on duplicate code can be ignored for now.

**Comment by abilpraju-aot** [comment]
> SonarCloud warning on duplicate code can be ignored for now.

ok

File: forms-flow-web/src/components/Modeler/SubflowCreateEdit.js (hunk 3 above)
Comment (abilpraju-aot) [reply forms-flow-web/src/components/Modeler/SubflowCreateEdit.js:171]: here we handling two scenarios edit and create thats why called after response

**Review by fahad-aot — APPROVED** [review]

**Review by arun-s-aot — APPROVED** [review]

**Pull request merged**